internal/logic_handlers/organizationlogic/service_test.go

```diff
@@ -0,0 +1,674 @@
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/volatiletech/null/v8"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+	"strconv"
+	"testing"
+	"time"
+)
+
+var (
+	mockTimeout      = 5 * time.Second
+	mockLogger       = logger.New(context.TODO())
+	mockDbReturnsErr = "error getting db"
+)
+
+type dependencies struct {
+	Persistor  persistor
+	Logger     *logrus.Entry
+	TxProvider persistence.TransactionProvider
+	Db         *sqlx.DB
+	Cleanup    func(ignoreErrors ...bool)
+}
+
+func getConcreteDependencies(t *testing.T) (*dependencies, func(ignoreErrors ...bool)) {
+	db, cp, cleanup := mysqlhelper.TestGetMockMariaDB(t)
+
+	store, err := mysqlstore.New(&mysqlstore.Config{
+		Logger: mockLogger,
+		QueryTimeouts: &persistence.QueryTimeouts{
+			Query: mockTimeout,
+			Exec:  mockTimeout,
+		},
+	})
+	require.NoError(t, err, "unexpected new mysqlstore error")
+
+	tx, err := mysqltx.New(&mysqltx.Config{
+		Logger:       mockLogger,
+		Db:           db,
+		DatabaseName: cp.Database,
+	})
+	require.NoError(t, err, "unexpected new mysqltx error")
+
+	prov, err := mysqlconn.New(&mysqlconn.Config{
+		Logger:    mockLogger,
+		TxHandler: tx,
+	})
+	require.NoError(t, err, "unexpected new mysqlconn error")
+
+	return &dependencies{
+		Persistor:  store,
+		TxProvider: prov,
+		Logger:     mockLogger,
+		Cleanup:    cleanup,
+		Db:         db,
+	}, cleanup
+}
+
+type argsGetOrganizations struct {
+	ctx    context.Context
+	filter *model.OrganizationFilters
+}
+
+type testCaseGetOrganizations struct {
+	name            string
+	getDependencies func(t *testing.T) (*dependencies, func(ignoreErrors ...bool))
+	args            argsGetOrganizations
+	mutations       func(t *testing.T, db *sqlx.DB)
+	assertions      func(t *testing.T, organization *model.PaginatedOrganizations, err error)
+}
+
+func getGetOrganizationTestCases() []testCaseGetOrganizations {
+	return []testCaseGetOrganizations{
+		{
+			name:            "success",
+			getDependencies: getConcreteDependencies,
+			args: argsGetOrganizations{
+				ctx:    context.Background(),
+				filter: &model.OrganizationFilters{},
+			},
+			mutations: func(t *testing.T, db *sqlx.DB) {
+				entry := mysqlmodel.Organization{
+					Name:          "Demby",
+					CreatedBy:     null.IntFrom(1),
+					LastUpdatedBy: null.IntFrom(1),
+				}
+
+				err := entry.Insert(context.Background(), db, boil.Infer())
+				require.NoError(t, err, "error inserting in the user db")
+			},
+			assertions: func(t *testing.T, paginated *model.PaginatedOrganizations, err error) {
+				require.NoError(t, err, "unexpected get organizations error")
+				require.NotNil(t, paginated, "unexpected nil organizations")
+				require.NotNil(t, paginated.Pagination, "unexpected nil pagination")
+				require.NotNil(t, paginated.Organizations, "unexpected nil organizations")
+				assert.NotEqual(t, 0, paginated.Pagination.RowCount, "unexpected total count")
```

Comment:
would be more accurate to explicitly list how many you expect to see VS checking if not 0